Gems/Atom/Feature/Common/Code/Source/DiffuseGlobalIllumination/DiffuseProbeGridBlendDistancePass.cpp

```diff
@@ -44,29 +43,35 @@ namespace AZ
 
         void DiffuseProbeGridBlendDistancePass::LoadShader()
         {
-            // load shader
-            // Note: the shader may not be available on all platforms
-            AZStd::string shaderFilePath = "Shaders/DiffuseGlobalIllumination/DiffuseProbeGridBlendDistance.azshader";
-            m_shader = RPI::LoadCriticalShader(shaderFilePath);
-            if (m_shader == nullptr)
+            // load shaders, each supervariant handles a different number of rays per probe
+            // Note: the raytracing shaders may not be available on all platforms
+            m_shaders.reserve(DiffuseProbeGridNumRaysPerProbeArraySize);
+            for (uint32_t index = 0; index < DiffuseProbeGridNumRaysPerProbeArraySize; ++index)
             {
-                return;
-            }
+                AZStd::string shaderFilePath = "Shaders/DiffuseGlobalIllumination/DiffuseProbeGridBlendDistance.azshader";
+                Data::Instance<RPI::Shader> shader = RPI::LoadCriticalShader(shaderFilePath, DiffuseProbeGridNumRaysPerProbeArray[index].m_supervariant);
```

Comment:
Is it worth trying to load this asynchronously now that there are a number of additional variants that will be needed?

Reply:
At some point we could look into that if load times become an issue.